src/doc/dynamic-throttle.txt

```diff
@@ -0,0 +1,129 @@
+     will again induce spiky behavior. The configs work very similarly to the
+     Filestore op worker queue throttle.
+
+        journal_throttle_low_threshhold
+	journal_throttle_high_threshhold
+	filestore_expected_throughput_ops
+	filestore_expected_throughput_bytes
+	journal_throttle_high_multiple
+	journal_throttle_max_multiple
+
+This scheme will not be inducing any delay between [0, low_threshold].
+In [low_threshhold, high_threshhold), delays should be injected based on a line
+from 0 at low_threshold to high_multiple * (1/expected_throughput) at high_threshhold.
+In [high_threshhold, 1), we want delays injected based on a line from
+(high_multiple * (1/expected_throughput)) at high_threshhold to 
+(high_multiple * (1/expected_throughput)) + (max_multiple * (1/expected_throughput)) 
+at 1.
+Now setting *_high_multiple and *_max_multiple to 0 should give us similar effect
+like old throttle. Setting filestore_queue_max_ops and filestore_queue_max_bytes
+to zero should disable the entire backoff throttle.
+          
+-------------------------
+HOW TO PICK PROPER VALUE ?
+-------------------------
+
+General guideline is the following.
+
+  filestore_queue_max_bytes and filestore_queue_max_ops:
+  -----------------------------------------------------
+
+  This directly depends on how filestore op_wq will be growing in the memory. So,
+  user needs to calculate how much memory he can give to each OSD. Bigger value
+  meaning current/max ratio will be smaller and throttle will be relaxed a bit.
+  This could help in case of 100% write but in case of mixed read/write this can
+  induce more latencies for reads on the objects those are not yet been applied 
+  i.e yet to reach in it's final filesystem location.
+
+  filestore_expected_throughput_bytes and filestore_expected_throughput_ops:
+  -------------------------------------------------------------------------
+
+  This directly co-relates to how much delay needs to be induced per throttle.
+  Ideally, this should be a bit above (or same) your backend device throughput.
+
+  *_low_threshhold *_high_threshhold :
+  ----------------------------------
+
+  If your backend is fast you may want to set this threshold value higher so that
+  initial throttle would be less and it can give a chance to back end transaction
+  to catch up.
+
+  *_high_multiple and *_max_multiple:
+  ----------------------------------
+
+  This is the delay tunables and one should do all effort here so that the current
+  doesn't reach to the max. For HDD and smaller block size, the old throttle may
+  make sense, in that case we should set these delay multiple to 0.
+
+It is recommended that user should use ceph_smalliobench tool to see the effect 
+of these parameters (and decide) on your HW before deploying OSDs. 
+Following scripts is a wrapper around small_io_bench and shows a nice latency 
+distribution.
+
+  https://github.com/athanatos/ceph-perf-scripts
```

Comment:
I really don't want this in the docs, it would be very confusing to users.  I'd rather you explained how to use the relevant smalliobenchfs options directly.  If you want, you can factor out the code for summarizing the smalliobenchfs output and put it into ceph.git as another file in another PR.